src/goto-programs/goto_clean_expr.cpp

```diff
@@ -363,16 +363,20 @@ void goto_convertt::clean_expr(
         expr.operands().size() == 2,
         "side-effect assignment expressions must have two operands");
 
-      if(expr.op1().id()==ID_side_effect &&
-         to_side_effect_expr(expr.op1()).get_statement()==ID_function_call)
+      auto &side_effect_assign = to_side_effect_expr_assign(expr);
+
+      if(
+        side_effect_assign.rhs().id() == ID_side_effect &&
+        to_side_effect_expr(side_effect_assign.rhs()).get_statement() ==
+          ID_function_call)
       {
-        clean_expr(expr.op0(), dest, mode);
-        exprt lhs=expr.op0();
+        clean_expr(side_effect_assign.lhs(), dest, mode);
+        exprt lhs = side_effect_assign.lhs();
```

Comment:
How about doing `const auto &side_effect_assign = ...` above and then `exprt lhs = side_effect_assign.lhs();` followed by `clean_expr(lhs, dest, mode);` in here?

Reply:
done